core/kg-query/src/main/java/it/unibz/inf/ontop/query/translation/impl/RDF4JValueExprTranslator.java

```diff
@@ -396,4 +465,28 @@ public VariableOrGroundTerm translateRDF4JVar(Var v, boolean leafNode) {
                 : Optional.ofNullable(externalBindings.get(var))
                         .orElseGet(termFactory::getNullConstant);
     }
+
+    public static class ExtendedTerm {
+        private final ImmutableTerm term;
+        private final ImmutableMap<Variable, ValueExpr> existsMap ;
```

Comment:
Change type to `ImmutableMap<Variable, Exists>` ? I have seen many unchecked casts in the code.